### PR TITLE
Resolve default value breaking empty context form round trip

### LINKED ISSUE #49: `jenkins.model.Jenkins` default value breaks empty context form round trip
https://github.com/jenkinsci/extension-filter-plugin/blob/33202c8708d46c2e57466ecbb4a2d7e10620363c/src/main/java/org/jenkinsci/plugins/Exclusion.java#L23 and https://github.com/jenkinsci/extension-filter-plugin/blob/33202c8708d46c2e57466ecbb4a2d7e10620363c/src/main/java/org/jenkinsci/plugins/ConfigurableExtensionFilter.java#L92 indicate it should be possible to have a `null` context.

https://github.com/jenkinsci/extension-filter-plugin/blob/33202c8708d46c2e57466ecbb4a2d7e10620363c/src/main/resources/org/jenkinsci/plugins/ConfigurableExtensionFilter/global.jelly#L12 breaks round-trip behavior, as it will replace a `null` value with `jenkins.model.Jenkins` on subsequent form submissions.

`JenkinsRule#configRoundtrip` can help write a test.

### PR DESCRIPTION
**Pull request by sridamul**
Closes #49 

I am not sure about removing the default value in the context field. I thought one can easily understand about context with the help of `help-context.html`

cc: @jonesbusy

### CODE REVIEW

**Comment by jonesbusy** [comment]
Hi,

Looks good to me and thanks for adding a test.

Did you also do some interactive testing ? The plugin lacks also coverage to be 100% confident of the changes

Thanks

**Review by jonesbusy — APPROVED** [review]

**Comment by sridamul** [comment]
Ok! Let me do that in the next PR

**Comment by sridamul** [comment]
> Did you also do some interactive testing ? 

Yes, I did and seems to work fine as before

File: src/test/java/org/jenkinsci/plugins/ConfigurableExtensionFilterTest.java [hunk 1]
```diff
@@ -23,4 +24,14 @@ public void cannotSetNull() {
...
     }
+
+    @Test
+    public void testConfigRoundtrip() throws Exception {
+        DescriptorImpl impl = new DescriptorImpl();
+        impl.setExclusions(new Exclusion[0]);
+
+        j.configRoundtrip();
+
+        assertEquals(0, impl.getExclusions().length);
```
Comment (daniel-beck) [inline src/test/java/org/jenkinsci/plugins/ConfigurableExtensionFilterTest.java:35]: This assertion does not fail without the change to `src/main`, so is unrelated to the issue the `src/main` change addresses.

File: src/main/resources/org/jenkinsci/plugins/ConfigurableExtensionFilter/global.jelly [hunk 2]
```diff
@@ -9,7 +9,7 @@
             <f:textbox name="fqcn"/>
           </f:entry>
           <f:entry title="${%Context}" field="context">
-            <f:textbox default="jenkins.model.Jenkins"/>
+            <f:textbox default="${null}"/>
```
Comment (daniel-beck) [inline src/main/resources/org/jenkinsci/plugins/ConfigurableExtensionFilter/global.jelly:12]: Easier as
```suggestion
            <f:textbox/>
```

**Pull request merged**